### PR TITLE
Optimization md_update_all_neighbour_arrays_multiple() by 30%

### PR DESCRIPTION
**Pull request by spawlows**
from 1.5% to 1.0% of all encodeing in M6.

### CODE REVIEW

**Comment by 1480c1** [comment]
> MAC OS Build fails

https://github.com/OpenVisualCloud/SVT-AV1/issues/132#issuecomment-468965545
Maybe this might help point out why it is failing.

**Comment by hassount** [comment]
@1480c1 
Thank you, for the feedback. Looked into the differences, this one seems to be failing for a different reason.
@spawlows 
Can you please look into the build failure here

**Comment by hassount** [comment]
@joelsole @andrey-norkin @kylophone @mafonsoN 
Can we move forward with this PR?

File: Source/Lib/Common/Codec/EbNeighborArrays.c [hunk 1]
```diff
@@ -825,9 +791,10 @@ void neighbor_array_unit_mode_write(
         count = block_width >> na_unit_ptr->granularityNormalLog2;
 
         for (idx = 0; idx < count; ++idx) {
-
-            EB_MEMCPY(dst_ptr, value, naUnitSize);
-
+            /* memcpy less that 10 bytes*/
+            for (j = 0; j < naUnitSize; ++j) {
+                dst_ptr[j] = value[j];
+            }
```
Comment (joelsole) [inline Source/Lib/Common/Codec/EbNeighborArrays.c]: braces not needed

File: Source/Lib/Common/Codec/EbNeighborArrays.c [hunk 2]
```diff
@@ -860,9 +827,10 @@ void neighbor_array_unit_mode_write(
         count = block_height >> na_unit_ptr->granularityNormalLog2;
 
         for (idx = 0; idx < count; ++idx) {
-
-            EB_MEMCPY(dst_ptr, value, naUnitSize);
-
+            /* memcpy less that 10 bytes*/
+            for (j = 0; j < naUnitSize; ++j) {
+                dst_ptr[j] = value[j];
+            }
```
Comment (joelsole) [inline Source/Lib/Common/Codec/EbNeighborArrays.c]: braces.
Same in other occurrences below

File: Source/Lib/Common/Codec/EbNeighborArrays.c [hunk 3]
```diff
@@ -901,9 +869,10 @@ void neighbor_array_unit_mode_write(
         count = ((block_width + block_height) >> na_unit_ptr->granularityTopLeftLog2) - 1;
 
         for (idx = 0; idx < count; ++idx) {
-
-            EB_MEMCPY(dst_ptr, value, naUnitSize);
-
+            /* memcpy less that 10 bytes*/
+            for (j = 0; j < naUnitSize; ++j) {
+                dst_ptr[j] = value[j];
+            }
```
Comment (joelsole) [inline Source/Lib/Common/Codec/EbNeighborArrays.c]: braces

File: Source/Lib/Common/Codec/EbNeighborArrays.h [hunk 4]
```diff
@@ -89,21 +89,36 @@ extern "C" {
...
+        uint32_t               loc_y) {
+        return (loc_y >> na_unit_ptr->granularityNormalLog2);
+    }
+
+    /*************************************************
+     * Neighbor Array Unit Get Top Index
+     *************************************************/
+    static INLINE uint32_t get_neighbor_array_unit_top_index32(
+        NeighborArrayUnit32_t *na_unit_ptr,
+        uint32_t               loc_x) {
```
Comment (joelsole) [inline Source/Lib/Common/Codec/EbNeighborArrays.h]: brace in next line, according to STYLE.md:

static void function(const int argument1,
                     const int argument2)
{
    do_something();
}

ditto below

File: Source/Lib/Common/Codec/EbPictureOperators.h [hunk 5]
```diff
@@ -168,15 +168,29 @@ extern "C" {
...
+    static INLINE void memset32bit(
+        uint32_t                     * in_ptr,
+        uint32_t                       value,
+        uint64_t                       num_of_elements)
+    {
+        uint64_t i;
+
+        for (i = 0; i < num_of_elements; i++) {
+            in_ptr[i] = value;
+        }
```
Comment (joelsole) [inline Source/Lib/Common/Codec/EbPictureOperators.h]: no curly braces according to STYLE.md

File: Source/Lib/Common/Codec/EbPictureOperators.h [hunk 6]
```diff
@@ -168,15 +168,29 @@ extern "C" {
...
-        uint64_t  num_of_elements);
+    static INLINE void memset16bit(
+        uint16_t                     * in_ptr,
+        uint16_t                       value,
+        uint64_t                       num_of_elements)
+    {
+        uint64_t i;
+
+        for (i = 0; i < num_of_elements; i++) {
+            in_ptr[i] = value;
```
Comment (joelsole) [inline Source/Lib/Common/Codec/EbPictureOperators.h]: no curly braces according to STYLE.md

**Pull request merged**